googlesearch/__init__.py

```diff
@@ -165,40 +166,33 @@ def get_page(url, user_agent=None, verify_ssl=True):
     :raises urllib2.URLError: An exception is raised on error.
     :raises urllib2.HTTPError: An exception is raised on error.
     """
-    if user_agent is None:
-        user_agent = USER_AGENT
+    user_agent = USER_AGENT if user_agent is None else user_agent
     request = Request(url)
     request.add_header('User-Agent', user_agent)
     cookie_jar.add_cookie_header(request)
-    if verify_ssl:
-        response = urlopen(request)
-    else:
-        context = ssl._create_unverified_context()
-        response = urlopen(request, context=context)
+    :
+    response = urlopen(request) if verify_ssl urlopen(request, context=ssl._create_unverified_context())
     cookie_jar.extract_cookies(response, request)
     html = response.read()
     response.close()
     try:
         cookie_jar.save()
-    except Exception:
+    except Exception: <== # What exception gets raised here?
         pass
     return html
 
 
 # Filter links found in the Google result pages HTML code.
 # Returns None if the link doesn't yield a valid result.
-def filter_result(link):
+def filter_result(link:str):
+    o = urlparse(link, 'http')
```

Comment:
I moved this into a higher scope so it's not instantiated twice within the underlying code

Reply:
how is this ever instantiated twice inside this function?
also, you are potentially introducing a bug here by moving code inside a try-catch outside.